draft-regext-rdap-extensions.md

```diff
@@ -54,144 +54,236 @@ Domain Name Registries (DNRs), Regional Internet Registries
-(note the lowercase if "ic" in "Nic") is not allowed.
-
-# Usage in Queries {#usage_in_queries}
-
-[@!RFC9082, section 5] describes the use of extension identifiers in formulating
-URIs to query RDAP servers. The extension identifiers are to be prepended to the
-path segments they use. For example, if an extension uses the identifier
-"foobar", then the path segments used in that extension are prepended with "foobar_".
-If the "foobar" extension defines paths "fizz" and "fazz", the URIs for this
-extension might take the following form:
+# Identifiers {#identifiers}
+
+## Purpose {#purpose}
+
+[@!RFC7480, section 6] describes the identifier used to signify RDAP
+extensions and the IANA registry into which RDAP extensions are to be
+registered.
+
+When in use in RDAP, extension identifiers are prepended to URL path
+segments, URL query parameters, and JSON object member names (herein
+further referred to as "JSON names").  They are also included in the
+"rdapConformance" member of each response that relies on the
+extension, so that clients can determine the extensions being used by
+the server for that response.  The "/help" response returns an
+"rdapConformance" member containing the identifiers for all extensions
+used by the server.
+
+The main purpose of the extension identifier is to act as a namespace,
+preventing collisions between elements from different extensions.
+Additionally, implementers and operators can use the extension
+identifiers to find extension definitions via an IANA registry.
+
+### Profile and Marker Extensions {#profiles_and_markers}
+
+While the RDAP extension mechanism was created to extend RDAP queries
+and/or responses, extensions can also be used to signal server policy
+(for example, specifying the conditions of use for existing response
+structures). Extensions that are primarily about signaling server
+policy are often called "profiles".
+
+Some extensions exist to denote the usage of values placed into an
+IANA registry, such as the IANA RDAP registries, or the usage of
+extensions for specifications used in RDAP responses, such as extended
+vCard/jCard properties.  Such extensions exist to "mark" these usages
+and are often called "marker" extensions.
+
+For example, an extension may be used to signal desired processing of
+a "rel" attribute in a "links" array, where the "rel" value is
+registered in the Link Relations Registry
+(<https://www.iana.org/assignments/link-relations/link-relations.xhtml>):
+
+    {
+      "rdapConformance": [
+        "rdap_level_0",
+        "lunarNIC"
+      ],
+      "objectClassName": "domain",
+      "ldhName": "example.com",
+      "links": [
+        {
+          "value": "https://example.com/domain/example.com",
+          "href": "https://example.com/sideways_href",
+          "rel": "sideways",
+          "type": "application/rdap+json"
+        }
+      ]
+    }
+
+When defining the usage of link relations, extensions should specify
```

Comment:
'should', because many link relations are usable with arbitrary media
types.  Though maybe something like the following would work:

```
When defining the usage of link relations, extensions SHOULD specify
the media types expected to be used with those link relations at an
appropriate level of generality.  For example, the extension may
specify one or more complete media types (e.g. "text/html"), or all
subtypes of a type (e.g. "text/*"), or any media type (in instances
where there are no constraints on the expected media types that may be
used with the link relation).
```